File: 14. Longest Common Prefix/Solution14.java
class Solution14 {
    public static String longestCommonPrefix(String[] strs) {
        int min = Integer.MAX_VALUE;
        for (String string : strs) {
            if (string.length() < min) {
                min = string.length();
            }
        }
        int i = 0;
        while (i < min) {
            for (String str : strs) {
                if (str.charAt(i) != strs[0].charAt(i)) {
                    return str.substring(0, i);
                }
            }
            i++;
        }
        return strs[0].substring(0, i);
    }

    public static void main(String[] args) {
        String[] arr = { "ab", "a" };
        System.out.println(longestCommonPrefix(arr));

    }
}

File: 14. Longest Common Prefix/Solution14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        min = float("inf")
        for string in strs:
            if len(string) < min:
                min = len(string)
        
        i = 0
        while i < min:
            for string in strs:
                if (string[i] != strs[0][i]):
                    return string[:i]
            i += 1
        return strs[0][:i]    


c = Solution()

print(c.longestCommonPrefix([["flower","flow","flight"]]))